test/bin/manage_build_cache.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+#
+# This script should be run on the build host to manage cache of build artifacts
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck source=test/bin/common.sh
+source "${SCRIPTDIR}/common.sh"
+
+AWS_BUCKET_NAME="${AWS_BUCKET_NAME:-microshift-build-cache}"
+BCH_SUBDIR=
+TAG_SUBDIR=
+
+usage() {
+    cat - <<EOF
+${BASH_SOURCE[0]} (upload|download|exist|latest|keep) [options]
+
+Manage build cache at the '${AWS_BUCKET_NAME}' AWS S3 bucket. The script
+assumes that the bucket exists and it is configured for read-write access
+using 'aws s3 (ls|sync|rm)' operations.
+
+The cache directory structure is '${AWS_BUCKET_NAME}/<branch>/<tag>'.
+
+  -h        Show this help.
+
+  upload:   Upload build artifacts from the local disk to the specified
+            '${AWS_BUCKET_NAME}/<branch>/<tag>' AWS S3 bucket.
+
+  download: Download build artifacts from the specified
+            '${AWS_BUCKET_NAME}/<branch>/<tag>' AWS S3 bucket to
+            the local disk.
+
+  verify:   Exit with 0 status if the specified
+            '${AWS_BUCKET_NAME}/<branch>/<tag>' sub-directory exists
+            and contains files, 1 otherwise.
+
+  latest:   Update the '${AWS_BUCKET_NAME}/<branch>/latest' directory
+            in the AWS S3 bucket to a copy of the specified '<tag>'.
```

Comment:
Instead of "latest" being a copy of the content, could it be a file that tells the download script which tag is the latest? Then it could be updated atomically and wouldn't cause issues by deleting content that other jobs (or users) might be in the middle of downloading.

Reply:
Probably, I was planning to work out the details when implementing the usage of this script. We can adopt it as we go, according to our needs.

Reply:
I implemented the use of `last` tag according to your suggestion. 
It seems to be much better